### PR TITLE
[jit] Emit script function calls during tracing.

### PR DESCRIPTION
**Pull request by suo**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#25089 [jit] Emit script function calls during tracing.**

Previously, when the tracer encountered a scripted function (or method), it
inlined the function into the graph. Now, we emit a CallFunction or
CallMethod node instead.

Differential Revision: [D16987936](https://our.internmc.facebook.com/intern/diff/D16987936)

### CODE REVIEW

File: test/test_jit.py [hunk 1]
```diff
@@ -10547,6 +10546,7 @@ def traced_fn(x):
         self.assertTrue(len(list(traced_fn.graph.inputs())) == 1)
         FileCheck().check("aten::mm").check("aten::add").run(str(traced_fn.graph))
 
+    @_tmp_donotuse_dont_inline_everything
```
Comment (zdevito) [inline test/test_jit.py]: but it says donotuse

File: torch/jit/__init__.py [hunk 2]
```diff
@@ -565,6 +565,8 @@ def graph_diagnostic_info():
                     break  # Graphs have already diverged
 
                 if n_mod.kind() == 'prim::Constant' and not (n_mod.mustBeNone() or n_check.mustBeNone()):
+                    if not n_mod.hasAttribute('value'):
```
Comment (zdevito) [inline torch/jit/__init__.py]: why did this change?

**Review by zdevito — APPROVED** [review]
Looks good!

File: test/test_jit.py (hunk 1 above)
Comment (suo) [reply test/test_jit.py]: :'(

**Pull request closed without merging**